host/sgx/create.c

```diff
@@ -401,6 +402,41 @@ static oe_result_t _initialize_enclave(oe_enclave_t* enclave)
     return result;
 }
 
+/*
+** _config_enclave()
+**
+** Config the enclave with an array of configurations.
+*/
+static oe_result_t _configure_enclave(
```

Comment:
Future work: Consider minimizing the number of calls between host an enclave as these transitions are expensive, and sets up a pattern for host to call into the enclave for _each_ type of configuration separately. Switchless OCALLs is a bit of a special case since most of this is on the host side, but it might be worth breaking that out so the setup can be done before enclave initialization, and then the manager handle passed to enclave as part of the `_initialize_enclave` call. I'd also imagine that for switchless ECALLs, we'd pass the config struct into the enclave as part of `_initialize_enclave` instead of via yet another separate config ECALL.

Reply:
Good point. I considered the option of using `_initialize_enclave` and its corresponding ECALL `OE_ECALL_INIT_ENCLAVE` to initialize switchless. But there are limitations for that:

* The function `_initialize_enclave` simply sets a opaque handle to the enclave created by the host. Therefore even if we set the switchess configurations, such as the switchless manager, in `oe_enclave_t` and pass the enclave instance to this function, it still can't access the switchless configurations.  
* We could combine the ECALLs that have to be made during `_initialize_enclave` and `configure_enclave`. This depends on that we have a full view of the `oe_enclave_t` instance inside the enclave. Not sure whether the opaque view is by design.